libs/designer-ui/src/lib/azureResourcePicker/index.tsx

```diff
@@ -39,20 +49,46 @@ export const AzureResourcePicker = (props: AzureResourcePickerProps) => {
     fetchSubResourcesCallback,
   } = props;
 
+  const intl = useIntl();
+
   const itemsQuery = useQuery([resourceType], async () => getResourcesCallback?.() ?? [], {
     enabled: true,
     staleTime: 1000 * 60 * 60 * 24,
   });
 
   const resources = useMemo(() => ((itemsQuery?.data ?? []) as any[]).sort((a, b) => a.name.localeCompare(b.name)), [itemsQuery.data]);
+  const resourceNames = useMemo(() => resources.map((resource) => resource.name), [resources]);
 
   const gridTemplateColumns = useMemo(() => '2fr '.repeat(headers.length - 1) + '1fr', [headers.length]);
 
+  const [searchTerm, setSearchTerm] = useState<string | null>(null);
+  const searchText = intl.formatMessage({
+    defaultMessage: 'Search',
+    description: 'Search box placeholder text',
+  });
+  const fuseObject = useMemo(
+    () =>
+      new Fuse(
+        resourceNames.map((name) => ({ text: labelCase(name), id: name })),
+        fuseOptions
+      ),
+    [resourceNames]
+  );
+  const searchResourceNames = useMemo(() => {
+    if (!searchTerm) return resourceNames;
+    return fuseObject.search(searchTerm).map((result) => result.item.id);
+  }, [resourceNames, fuseObject, searchTerm]);
+
+  const searchFilteredResources = useMemo(() => {
+    return resources.filter((resource) => searchResourceNames.includes(resource.name));
+  }, [resources, searchResourceNames]);
+
   return (
     <div style={{ width: '100%' }}>
       <Label className="label" required>
         {titleText}
       </Label>
+      <SearchBox placeholder={searchText} onChange={(e, newValue) => setSearchTerm(newValue ?? null)} />
```

Comment:
this is great, we should also have a load more kind of UI, since sometimes the resources can be big list

Reply:
The <List/> fluent component is virtualized, so it automatically does render paging under the hood, no real need for an explicit "load more"